### ISSUE TITLE
Manually calling formik.submitForm() resolves with undefined

### ISSUE THREAD

**Issue by Wgil** [formium/formik#2057]
### Current Behavior

Calling `formik.submitForm()` promise manually, resolves always with `undefined` regardless of what my handler returns.

### Desired Behavior

I'd like this to be resolved with the return value of my submit handler so I can decide what to do next.

https://codesandbox.io/s/inspiring-fog-9pmo8

**Comment by alexlurvey** [comment]
The issue I just had with this function was the promise not being rejected when the form is invalid.

The docs say this: The promise will be rejected if form is invalid.

I'm trying to figure out how to take action after an invalid submit without watching all the Formik state flags changing.

**Comment by jsilvax** [comment]
> The issue I just had with this function was the promise not being rejected when the form is invalid.
> 
> The docs say this: The promise will be rejected if form is invalid.
> 
> I'm trying to figure out how to take action after an invalid submit without watching all the Formik state flags changing.

I have the same issue that was closed as "fixed" (It's not...)  
https://github.com/jaredpalmer/formik/issues/1904#issuecomment-549113346

There's an open PR to fix it, but sadly has not been reviewed  
https://github.com/jaredpalmer/formik/pull/1843

**Comment by DamodarSojka** [comment]
Also run into that, for anyone looking for a temp workaround:
use validateForm to check if there are any errors

**Comment by jaredpalmer** [comment]
Fixed in 2.1

**Comment by chornos13** [comment]
I still experienced with the issues I don't know what is wrong

**Comment by shakibmiraki** [comment]
I experienced the issues too 
the version i'am using is 2.1.4

**Comment by druvisc** [comment]
Also getting `undefined` on `2.1.4`.

**Comment by gastonmorixe** [comment]
Same, since combinedErrors is not an Error it doesn't throw.

**Comment by maddhruv** [comment]
re-opening as one more issue is reported - https://github.com/formium/formik/issues/2831

**Comment by Macilias** [comment]
there is still no solution for that =0_0=

**Comment by dilshankelsen9y** [comment]
Running into this issue as well

**Comment by trungdeveloper** [comment]
Is there a workaround?

**Comment by Uvir127** [comment]
Has a workaround or fix been found for this?

**Comment by HammadRafiq** [comment]
I faced the same problem and got it fixed with my own workaround. I am validating and submitting the form on a manual click:
                                onClick={() => {
                                    formRef.current.submitForm() 
                                    formRef.current.validateForm() 
                                        .then(values => {
                                            if (Object.keys(values).length === 0) {
                                                onSubmit(formRef.current.values)
                                            }
                                        })
                                }}
onSubmit() is the function that I only want to call if every value in my form is valid. formRef.current.submitForm() displays the errors (if there are any). The if condition inside formRef.current.validateForm() will only be true if there are no errors in the form.

In a nutshell:
1. If some field is invalid in the form, show the error and dont call the onSubmit() function
2. If all the data in the form is valid, call the onSubmit() function

formRef.current.values gives me all the values of form. If you are not aware, read it here on how to access Formik properties using useRef:
https://stackoverflow.com/questions/73233138/how-to-submit-formik-when-button-submit-outside-tag-formik-in-reactjs
